### ISSUE TITLE
Running executables sporadically fails with ETXTBSY (Text file busy)

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle#31282]
### Current Behavior

Running executables concurrently (with Exec, project.exec, ExecOperations.exec plus the JavaExec versions) can fail sporadically if the Gradle daemon process has recently written a file. 

### Expected Behavior

This should work

### Context (optional)

In https://github.com/gradle/gradle/pull/30269, I removed the use of native-platform's `ProcessLauncher` in favor of the JDK's built-in `ProcessBuilder`. 
On non-Windows platforms, we were already using an implementation of `ProcessLauncher` that used `ProcessBuilder`; however, `ProcessLauncher` also had a workaround to only launch a single executable at a time:
https://github.com/gradle/native-platform/blob/master/native-platform/src/main/java/net/rubygrapefruit/platform/internal/WrapperProcessLauncher.java#L33-L37

This appears to be related to a [Java platform bug](https://bugs.openjdk.org/browse/JDK-8068370) which is a common problem across other programs on Linux/Unix.  [Go has a good explanation of the problem](https://github.com/golang/go/issues/22315).

### Self-contained Reproducer Project

You can reproduce this outside of Gradle with this Groovy script:
```
def threads = []
1000.times { idx ->
  threads << new Thread({
    def script = new File("script$idx")
    script.text = """#!/bin/bash
    echo "Hello $it"
    """
    script.executable = true
    def pb = new ProcessBuilder(script.absolutePath)
    def p
    //synchronized(Object.class) {
       p = pb.start()
    //}
    p.waitFor()
    script.delete()
  })
}
println("starting")
threads.each { it.start() }
println("waiting")
threads.each { it.join() }
println("done")
```

Putting the synchronized block back in "fixes" it. 

### Gradle version

8.11

### Gradle version that used to work

<8.11

**Comment by liutikas** [comment]
Any updates here? We'd really love a fix for this so we can adopt 8.11.*

**Comment by ljacomet** [comment]
Fixed with:
* #31283
